multipool-tracker/src/multipool_storage/read.rs

```diff
@@ -1,14 +1,18 @@
 use std::ops::Shl;
 
-use super::{expiry::MayBeExpired, merge, Multipool, MultipoolAsset, Price, Share, X32, X96};
+use super::{
+    errors::MultipoolErrors, expiry::MayBeExpired, expiry::Merge, merge, Multipool, MultipoolAsset,
+    Price, Share, X32, X96,
+};
 use ethers::prelude::*;
 
 impl Multipool {
-    pub fn asset(&self, asset_address: &Address) -> Option<MultipoolAsset> {
+    pub fn asset(&self, asset_address: &Address) -> Result<MultipoolAsset, MultipoolErrors> {
         self.assets
             .iter()
             .find(|asset| asset.address.eq(asset_address))
             .cloned()
+            .ok_or(MultipoolErrors::PriceNotFound(asset_address))
```

Comment:
* Price not found doesn't seems to be an error for missing of a particular asset
* Returning ```Option<MultipoolAsset>``` here is better way as there is no error kinds and function can't error at all. It can only return the information that there is no such asset which is idiomatically an ```Option``` value. Later on whoever uses this function can convert this lacking into an error on their demand